v2/pkg/core/engine.go

```diff
@@ -0,0 +1,57 @@
+package core
+
+import (
+	"github.com/projectdiscovery/nuclei/v2/pkg/protocols"
+	"github.com/projectdiscovery/nuclei/v2/pkg/types"
+)
+
+// Engine is an engine for running Nuclei Templates/Workflows.
+//
+// The engine contains multiple thread pools which allow using different
```

Comment:
What do you mean by different concurrency values?

Reply:
Meaning that separate concurrency values can be provider for headless and other templates concurrency.

Reply:
So you meant concurrency configuration then